fast_matched_filter/src/matched_filter.c

```diff
@@ -95,6 +97,8 @@ float network_corr(float *templates, float *sum_square_template, int *moveouts,
             component_offset = station_offset + c;
             if (weights[component_offset] == 0) continue;
 
+            if ((i + moveouts[component_offset]) > n_samples_data - n_samples_template) continue;
```

Comment:
This condition does not need to be tested if we already make sure we stop correlating when the first channel goes out of the data stream.